src/main/java/com/aws/greengrass/logmanager/model/ProcessingFiles.java

```diff
@@ -45,6 +48,28 @@ public void put(LogManagerService.CurrentProcessingFileInformation value) {
         evictStaleEntries();
     }
 
+    /**
+     * Removes an entry from the cache for a provided file hash.
+     * @param fileHash - A file hash.
+     */
+    public void remove(String fileHash) {
+        Node node = cache.remove(fileHash);
```

Comment:
handle null filehash?

Reply:
Yeah we should. Theoretically that would not happen because we filter and not process empty file hashes but this calls should defensively handle it.